pkg/operator/reporting/templates.go

```diff
@@ -129,10 +130,32 @@ func (ctx *ReportQueryTemplateContext) newQueryTemplate() (*template.Template, e
 // and checks if the returned error is nil. If nil, return an empty string and the error, else return
 // the function call to renderTemplate, passing in the new query template and tmplCtx parameter.
 func RenderQuery(ctx *ReportQueryTemplateContext, tmplCtx TemplateContext) (string, error) {
+	// validate the ctx.ReportQuery.Spec.Query field and create a new template
 	tmpl, err := ctx.newQueryTemplate()
 	if err != nil {
 		return "", err
 	}
+
+	// validate that all required inputs were passed
+	expectedVals := sets.NewString()
+	expectedInputs := sets.NewString()
+
+	for _, queryDef := range ctx.ReportQuery.Spec.Inputs {
+		// check whether the current definition is required
+		if queryDef.Required {
+			expectedInputs.Insert(queryDef.Name)
+		}
+	}
+
+	for queryKey := range tmplCtx.Report.Inputs {
+		expectedVals.Insert(queryKey)
+	}
+
+	// check if there's a difference in input definitions and what's provided as an input value, and vice-versa
+	if len(expectedInputs.Difference(expectedVals)) != 0 || len(expectedVals.Difference(expectedInputs)) != 0 {
```

Comment:
Here you could actually return an error message including the list of inputs provided that aren't expected and vice-versa. Generally, try to have more specific error messages than generic ones like the one below.